latte/modules/callbacks.py

```diff
@@ -0,0 +1,61 @@
+"""
+Implementations of various callbacks used throughout `Latte`, either in training VAEs or MIST.
+"""
+
+import copy
+from collections import defaultdict
+
+import pytorch_lightning as pl
+from torch.utils.tensorboard import SummaryWriter
+
+from pythae.trainers import training_callbacks
+
+
+class MetricTracker(pl.Callback):
+    """A callback to track all the epoch-level metrics logged."""
+
+    def __init__(self):
+        self.epoch_outputs = []
+
+    def on_validation_epoch_end(self, trainer: pl.Trainer, module: pl.LightningModule) -> None:
+        self.epoch_outputs.append(copy.deepcopy(trainer.logged_metrics))
```

Comment:
https://twitter.com/gvanrossum/status/383618349000691712

Not that we need to fix this now, but in general `deepcopy` is discouraged. (In general, it's good to have as many immutable things as possible).